dotnet/src/SemanticKernel.Skills/Skills.Memory.Sqlite/SqliteMemoryStore.cs

```diff
@@ -2,80 +2,153 @@
 
 using System;
 using System.Collections.Generic;
-using System.Diagnostics.CodeAnalysis;
+using System.Collections.ObjectModel;
+using System.Data;
+using System.Data.Common;
 using System.Globalization;
+using System.Linq;
 using System.Runtime.CompilerServices;
+using System.Text;
 using System.Text.Json;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Data.Sqlite;
+using Microsoft.SemanticKernel.AI.Embeddings;
+using Microsoft.SemanticKernel.AI.Embeddings.VectorOperations;
+using Microsoft.SemanticKernel.Memory;
+using Microsoft.SemanticKernel.Memory.Collections;
 using Microsoft.SemanticKernel.Memory.Storage;
 
 namespace Microsoft.SemanticKernel.Skills.Memory.Sqlite;
-
-/// <summary>
-/// An implementation of <see cref="IDataStore{TValue}"/> backed by a SQLite database.
-/// </summary>
-/// <remarks>The data is saved to a database file, specified in the constructor.
-/// The data persists between subsequent instances. Only one instance may access the file at a time.
-/// The caller is responsible for deleting the file.</remarks>
-/// <typeparam name="TValue">The type of data to be stored in this data store.</typeparam>
-public class SqliteDataStore<TValue> : IDataStore<TValue>, IDisposable
+public class SqliteMemoryStore<TEmbedding> : IMemoryStore<TEmbedding>, IDisposable
+    where TEmbedding : unmanaged
 {
-    /// <summary>
-    /// Connect a Sqlite database
-    /// </summary>
-    /// <param name="filename">Path to the database file. If file does not exist, it will be created.</param>
-    /// <param name="cancel">Cancellation token</param>
-    [SuppressMessage("Design", "CA1000:Do not declare static members on generic types",
-        Justification = "Static factory method used to ensure successful connection.")]
-    public static async Task<SqliteDataStore<TValue>> ConnectAsync(string filename,
-        CancellationToken cancel = default)
+    public SqliteMemoryStore(SqliteConnection dbConnection)
+    {
+        this._dbConnection = dbConnection;
+    }
+
+    /// <inheritdoc/>
+    public IAsyncEnumerable<(IEmbeddingWithMetadata<TEmbedding>, double)> GetNearestMatchesAsync(
+        string collection,
+        Embedding<TEmbedding> embedding,
+        int limit = 1,
+        double minRelevanceScore = 0)
     {
-        SqliteConnection dbConnection = await Database.CreateConnectionAsync(filename, cancel);
-        return new SqliteDataStore<TValue>(dbConnection);
+        if (limit <= 0)
+        {
+            return AsyncEnumerable.Empty<(IEmbeddingWithMetadata<TEmbedding>, double)>();
+        }
+
+        IAsyncEnumerable<DataEntry<IEmbeddingWithMetadata<TEmbedding>>> asyncEmbeddingCollection = this.TryGetCollectionAsync(collection);
+        IEnumerable<DataEntry<IEmbeddingWithMetadata<TEmbedding>>> embeddingCollection = asyncEmbeddingCollection.ToEnumerable();
+
+        bool bEmpty = embeddingCollection?.Any() ?? false;
```

Comment:
I believe you want this to be !Any(). If embeddingCollection is not null and has items, this line will mark bEmpty as true when the collection is not empty.